ta4j-core/src/main/java/org/ta4j/core/indicators/DPOIndicator.java

```diff
@@ -69,19 +71,22 @@ public DPOIndicator(BarSeries series, int barCount) {
      */
     public DPOIndicator(Indicator<Num> price, int barCount) {
         super(price);
-        this.barCount = barCount;
-        timeShift = barCount / 2 + 1;
-        this.price = price;
-        sma = new SMAIndicator(price, this.barCount);
+        int timeFrame = barCount / 2 + 1;
+        final SMAIndicator simpleMovingAverage = new SMAIndicator(price, barCount);
+        final PreviousValueIndicator previousSimpleMovingAverage = new PreviousValueIndicator(simpleMovingAverage,
+                timeFrame);
+
+        this.indicatorMinusPreviousSMAIndicator = new DifferenceIndicator(price, previousSimpleMovingAverage);
+        this.name = String.format("%s barCount: %s", getClass().getSimpleName(), barCount);
     }
 
     @Override
     protected Num calculate(int index) {
-        return price.getValue(index).minus(sma.getValue(index - timeShift));
+        return indicatorMinusPreviousSMAIndicator.getValue(index);
     }
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() + " barCount: " + barCount;
+        return name;
```

Comment:
change to `return String.format("%s barCount: %s", getClass().getSimpleName(), barCount);`

Reply:
why not caching this value that may be requested several times in loggers or other output data

Reply:
1. we should be consistent: all other indicators, as far as I know, do not cache value only because of the logger
2. why caching value that may be not used at all if one decides not to log
3. logging is always optional, no need to add overhead in constructor

Reply:
I agree, but in this case we would have to store the barCount only for this purpose in an extra field. Therefore I thought why not store the indicator name instead.

Reply:
Ok, then leave it as is.